docs/wiki/Self-Hosted.md

```diff
@@ -0,0 +1,56 @@
+
+For more information about using GitHub Actions self-hosted runners see, [About self-hosted runners](https://docs.github.com/actions/hosting-your-own-runners/about-self-hosted-runners).
+For more information about using Azure DevOps self-hosted agents see, [Azure Pipelines agents](https://docs.microsoft.com/azure/devops/pipelines/agents/agents?view=azure-devops&tabs=browser).
+
+## Virtual Machine Scale Sets (VMSS)
+
+Virtual Machine Scale Sets are optimal for hosting your self-hosted agents/runners. They are easy to create and manage and will automatically scale as resource demand changes. To learn more about Virtual Machine Scale Sets see, [Virtual Machine Scale Sets documentation](https://docs.microsoft.com/azure/virtual-machine-scale-sets/).
+
+## Image
+
+To setup a VMSS for your self-hosted agents/runners you need an image with all the required software installed. The pipelines and actions provided in the [AzOps-Accelerator](https://github.com/Azure/AzOps-Accelerator) repository are using the GitHub Actions Virtual Environments [`ubuntu-20.04`](https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-Readme.md) image by default. The images includes multiple tools and software including all AzOps requirements.
+
+### Build an image using GitHub Actions Virtual Environments
+
+If you want to use the same image as the hosted environments for your self-hosted agents/runners you can build an image following the guides in the [GitHub Actions Virtual Environments](https://github.com/actions/virtual-environments).
+
+### Building your own custom image
+
+If you want to build your own custom lightweight image the following software is required for AzOps to function properly.
+
+#### Required software
+
+AzOps have a couple of dependencies that are required to function properly. When building your custom image, make sure that the following tools are included in the image:
+
+| Software | Recommended minimum version |
+|:--|--|
+| `jq` | `1.6` |
+| `Git` | `2.35.1` |
+| `Azure CLI` | `2.34.1` |
+| `PowerShell` | `7.2.2` |
+| `Bicep` | `v0.5.6` |
+| `Github CLI` | `2.7.0` |
+
+> NOTE: `Bicep` is only required if you plan to deploy [Bicep](https://github.com/Azure/bicep) templates using AzOps.
+
+##### PowerShell Modules
+
+AzOps depends on some of the `Az` modules and `PSFramework`, these modules will be installed with `AzOps` during pipeline/action run and are not required to include in your image. But if you want to make your pipelines/actions as fast as possible the modules can be included in your image.
+
+| Module | Required version |
+|:--|--|
+| `Az.Accounts` | `2.7.3` |
+| `Az.Billing` | `2.0.0` |
+| `Az.Resources` | `5.4.0` |
+| `PSFramework` | `1.6.214` |
```

Comment:
To avoid maintaining the versions of the dependencies in yet another place, we could perhaps reference the module manifest instead? https://github.com/Azure/AzOps/blob/main/src/AzOps.psd1#L54